daphne/src/vdaf/mod.rs

```diff
@@ -839,52 +850,137 @@ impl VdafConfig {
             )
             .await?;
 
-        for initialized_report in initialized_reports.into_iter() {
-            let transition = match initialized_report {
-                EarlyReportStateInitialized::Ready {
-                    metadata,
-                    public_share: _,
-                    state,
-                    message,
-                } => {
-                    states.push((state, metadata.time, metadata.id.clone()));
-                    Transition {
-                        report_id: metadata.into_owned().id,
-                        var: TransitionVar::Continued(message.get_encoded()),
-                    }
-                }
+        Ok(initialized_reports)
+    }
 
-                EarlyReportStateInitialized::Rejected { metadata, failure } => {
-                    metrics.report_inc_by(&format!("rejected_{failure}"), 1);
-                    Transition {
-                        report_id: metadata.into_owned().id,
-                        var: TransitionVar::Failed(failure),
+    /// Consume an initial aggregate request from the Leader. The outputs are the Helper's state
+    /// for the aggregation flow and the aggregate response to send to the Leader. This method is
+    /// run by the Helper.
+    pub(crate) fn handle_agg_job_init_req(
```

Comment:
I think we should just split this function in two, one for each version of the protocol, even if that means having a bit of repeated code. Otherwise they get incredibly long and it's harder to follow when debugging because half the code doesn't matter (usually when debugging you only care about one version of the draft). It also makes it easier to rip out draft2 later and makes it easier to compare the 2 drafts.

Reply:
Great idea, done.